runtime/src/main/java/org/corfudb/runtime/view/TableRegistry.java

```diff
@@ -203,25 +232,36 @@ void registerTable(@Nonnull String namespace,
             // Or no modification to the protobuf files.
             try {
                 this.runtime.getObjectsView().TXBuild().type(TransactionType.WRITE_AFTER_WRITE).build().begin();
-                boolean hasSchemaChanged = false;
                 CorfuRecord<TableDescriptors, TableMetadata> oldRecord = this.registryTable.get(tableNameKey);
-                if (oldRecord != null && !oldRecord.getPayload().getFileDescriptorsMap()
-                        .equals(tableDescriptors.getFileDescriptorsMap())) {
-                    hasSchemaChanged = true;
-                    log.warn("registerTable: Schema update detected for table {}${}", namespace, tableName);
-
-                    for (StackTraceElement st : Thread.currentThread().getStackTrace()) {
-                        log.warn("{}", st);
+                if (oldRecord != null) {
+                    for (Map.Entry<ProtobufFileName, CorfuRecord<ProtobufFileDescriptor, TableMetadata>> entry :
+                            allDescriptors.entrySet()) {
+                        CorfuRecord<ProtobufFileDescriptor, TableMetadata> prevDescriptor = protobufDescriptorTable.get(entry.getKey());
+                        if (prevDescriptor != null) {
+                            // Do not attempt to compare default protobuf files which are huge & don't affect schemas.
+                            if (!entry.getKey().getFileName().startsWith("google/protobuf") &&
+                                    !prevDescriptor.getPayload().getFileDescriptor().equals(entry.getValue().getPayload().getFileDescriptor())) {
+                                log.warn("registerTable: Schema update detected for table {}${}", namespace, tableName);
+
+                                for (StackTraceElement st : Thread.currentThread().getStackTrace()) {
+                                    log.warn("{}", st);
+                                }
+                                log.debug("registerTable: old schema: {}", oldRecord.getPayload().getFileDescriptorsMap());
+                                log.debug("registerTable: new schema: {}", tableDescriptors.getFileDescriptorsMap());
+                                this.protobufDescriptorTable.put(entry.getKey(), entry.getValue());
+                                this.registryTable.put(tableNameKey,
+                                        new CorfuRecord<>(tableDescriptors, metadataBuilder.build()));
+                            }
+                        } else { // this is the first time we are seeing this new protobuf file descriptor
```

Comment:
When it reaches this else clause, oldRecord != null && prevDescriptor == null, which means it wants to update table schema using a new descriptor that we have never seen, right? If that's correct, then we can also add log messages here to log the schema change. BTW I found the nested if clauses kind of hard to understand.

Reply:
thanks - let me see if I can refactor this code section

Reply:
done refactoring, can you please review once more? thanks